Remove `qml.workflow.set_shots` (#6534)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h4>Other Improvements</h4>

* Added `qml.devices.qubit_mixed` module for mixed-state qubit device support. This module introduces:

  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:

  * Two density matrix contraction methods using `einsum` and `tensordot`

>   * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators

Reviewer: Looks like merge artifacts

> 
>   [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:
> 
>   * Two density matrix contraction methods using `einsum` and `tensordot`
> 
>   * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators

Author: ```suggestion
  - A new API for mixed-state operations
  - An `apply_operation` helper function featuring:
    - Two density matrix contraction methods using `einsum` and `tensordot`
    - Optimized handling of special cases including:
      - Diagonal operators
      - Identity operators 
      - CX (controlled-X)
      - Multi-controlled X gates
      - Grover operators
  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
```

<h3>Breaking changes 💔</h3>

> * The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
> workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.

Reviewer: ```suggestion
* The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface
  in our code. Instead, shots should be specified on the tape, and this should be used by the device.
```

<h3>Breaking changes 💔</h3>

> * The ``qml.workflow.set_shots`` helper function has been removed. We no longer interact with the legacy device interface
>   in our code. Instead, shots should be specified on the tape, and this should be used by the device.

Author: ```suggestion
* The ``qml.workflow.set_shots`` helper func has been removed. We no longer interact with the legacy device interface in our code 
  workflow behavior. Instead, shots should be specified on the tape, and the device should pull it's number of shots from the tape.
```

>   [(#6534)](https://github.com/PennyLaneAI/pennylane/pull/6534)

Reviewer: Same comment here.

Author: Just updated!
